reframe/frontend/statistics.py

```diff
@@ -105,15 +105,10 @@ def json(self, force=False):
                     'time_setup': t.duration('setup'),
                     'time_total': t.duration('total')
                 }
-                partition = check.current_partition
-                environ = check.current_environ
-                if partition:
-                    entry['system'] = partition.fullname
-                    entry['scheduler'] = partition.scheduler.registered_name
-
-                if environ:
-                    entry['environment'] = environ.name
-
+                partition = t.testcase.partition
```

Comment:
Better write it as follows and add a comment explaining why we are taking these values from the test case and not from the check:

```suggestion
                # We take partition and environment from the test case and not from the check, since if the test fails before `setup()`, these are not set inside the theck.
                partition = t.testcase.partition
                environ = t.testcase.environ
```